stdlib/public/runtime/BytecodeLayouts.cpp

```diff
@@ -1000,9 +1012,11 @@ static void blockCopy(const Metadata *metadata, LayoutStringReader1 &reader,
                       uintptr_t &addrOffset, uint8_t *dest, uint8_t *src) {
 #if SWIFT_OBJC_INTEROP
   uintptr_t _addrOffset = addrOffset;
-  auto *copy = _Block_copy(*(void**)(src + _addrOffset));
-  memcpy(dest + _addrOffset, &copy, sizeof(void*));
+  uintptr_t object = *(uintptr_t *)(src + _addrOffset);
+  memcpy(dest + _addrOffset, &object, sizeof(void *));
   addrOffset = _addrOffset + sizeof(void*);
+  object &= ~_swift_abi_SwiftSpareBitsMask;
+  _Block_copy((void *)object);
```

Comment:
Do we know for sure that the block has been copied to the heap already? `_Block_copy` returns its parameter iff the block isn't still on the stack. I recall that the Swift compiler doesn't try too hard to avoid copying blocks so we may never see a stack block here, but this worries me a bit, especially if the compiler got smarter about not copying nonescaping blocks at some point.